test/ussd_public.test.js

```diff
@@ -0,0 +1,677 @@
+                    .check.interaction({
+                        state: 'state_msisdn_not_recognised',
+                        reply: "We do not recognise this number. Please dial from the registered number or sign up with your local Community Health Extension worker."
+                    })
+                    .run();
+            });
+            // assuming flow via unregistered/registered user...
+            it("to state_main_menu (via state C)", function() {
+                return tester
+                    .setup.user.addr('082111')
+                    .inputs(
+                        {session_event: 'new'}  // dial in
+                        , '2'   // state_language - hausa
+                        , '0803304899'  // state_msg_registered_msisdn
+                    )
+                    .check.interaction({
+                        state: 'state_main_menu',
+                        reply: [
+                            "Select:",
+                            "1. Start Baby messages",
+                            "2. Change message preferences",
+                            "3. Change my number",
+                            "4. Change language",
+                            "5. Stop receiving messages"
+                        ].join('\n')
+                    })
+                    .run();
+            });
+            // registered user with permission to manage number
+            it("to state_main_menu (via state B)", function() {
+                return tester
+                    .setup.user.addr('082222')
+                    .inputs(
+                        {session_event: 'new'}  // dial in
+                        , '1'   // state_msisdn_permission - yes
+                    )
+                    .check.interaction({
+                        state: 'state_main_menu',
+                        reply: [
+                            "Select:",
+                            "1. Start Baby messages",
+                            "2. Change message preferences",
+                            "3. Change my number",
+                            "4. Change language",
+                            "5. Stop receiving messages"
+                        ].join('\n')
+                    })
+                    .run();
+            });
+            it("to state_msisdn_no_permission", function() {  // via state B
+                return tester
+                    .setup.user.addr('082222')
+                    .inputs(
+                        {session_event: 'new'}  // dial in
+                        , '2'   // state_msisdn_permission - yes
+                    )
+                    .check.interaction({
+                        state: 'state_msisdn_no_permission',
+                        reply: "We're sorry, you do not have permission to update the preferences for this subscriber."
+                    })
+                    .run();
+            });
+            it("to state_msg_already_registered_baby", function() {
```

Comment:
Since there's such a defined split in the flows that never really comes together again, I'd suggest using seperate 'describes' so it's easier to read the tests.